devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy

```diff
@@ -0,0 +1,392 @@
+	updateDataValue("fw", fw)
+	def text = "${device.displayName}: firmware version: $fw, Z-Wave version: ${cmd.zWaveProtocolVersion}.${cmd.zWaveProtocolSubVersion}"
+	createEvent(descriptionText: text, isStateChange: false)
+}
+
+def zwaveEvent(physicalgraph.zwave.Command cmd, endpoint = null) {
+	log.info "***** Unhandled Command called, cmd '${cmd}', endpoint '${endpoint}' *****"
+	[descriptionText: "Unhandled $device.displayName: $cmd", isStateChange: true]
+}
+
+
+/**
+ * PING is used by Device-Watch in attempt to reach the Device
+ * */
+def ping() {
+	log.info "ping called"
+	refresh()
+}
+
+def refresh(endpoint) {
+	log.info "refresh called: endpint '${endpoint}' "
+	if(endpoint) {
+		delayBetween([
+		secureEncap(zwave.basicV1.basicGet(), endpoint),
+		"delay 500"
+		], 500)
+	} else {
+		def commands = []
+		commands << zwave.sensorMultilevelV5.sensorMultilevelGet(sensorType: 1)
+		commands << zwave.sensorMultilevelV5.sensorMultilevelGet(sensorType: 5)
+		sendCommands(commands,1000)
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.configurationv2.ConfigurationReport cmd){
+	log.info "ConfigurationReport called: ${cmd}"
+	def result = []
+	def map = [:]
+	switch (cmd.parameterNumber) {
+		case 1:
+			state.tempOffset = cmd.scaledConfigurationValue
+			break
+		case 2:
+			state.humidityOffset = cmd.scaledConfigurationValue
+			break
+	}
+	log.debug "ConfigurationReport event: ${map}"
+	result <<  createEvent(name: "configReport", value: "${state.tempOffset}, ${state.humidityOffset}%")
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.multichannelv3.MultiChannelEndPointReport cmd, endpoint = null) {
+	log.info "MultiChannelEndPointReport called: cmd '${cmd}'"
+	if(!childDevices) {
+		if (isDawonWallSmartSwitch3()) {
+			addChildSwitches(3)
+		} else if (isDawonWallSmartSwitch2()) {
+			addChildSwitches(2)
+		} else if (isDawonWallSmartSwitch1()) {
+			addChildSwitches(1)
+		} else {
+			log.debug "child endpoint=$cmd.endPoints"
+			addChildSwitches(cmd.endPoints)
+		}
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.manufacturerspecificv2.ManufacturerSpecificReport cmd) {
+	log.info "ManufacturerSpecificReport called: cmd '${cmd}'"
+	log.debug "manufacturerId:   ${cmd.manufacturerId}"
+	log.debug "productId:        ${cmd.productId}"
+	log.debug "productTypeId:    ${cmd.productTypeId}"
+	def msr = String.format("%04X-%04X-%04X", cmd.manufacturerId, cmd.productTypeId, cmd.productId)
+	updateDataValue("MSR", msr)
+	createEvent([descriptionText: "$device.displayName MSR: $msr", isStateChange: false])
+}
+
+private sendCommands(cmds, delay=1000) {
+	log.info "sendCommands called: cmds '${cmds}', delay '${delay}'"
+	sendHubCommand(cmds, delay)
+}
+
+private commands(commands, delay=200) {
+	log.info "commands called: commands '${commands}', delay '${delay}'"
+	delayBetween(commands.collect{ command(it) }, delay)
+}
+
+private secure(cmd) {
+	log.info "secure called"
+	if(zwaveInfo.zw.endsWith("s")) {
```

Comment:
this (due to changes with S2) should now be `contains` instead of `endsWith`

Reply:
Thanks. This function ( secure ) will be patch.